feat: added pokemon api model

Miqqy0069 opened the pull request:
- added getPokemonList function

![image](https://user-images.githubusercontent.com/75653480/226325607-7d394287-58a1-4e2d-9001-551ec4f6530d.png)

Miqqy0069 committed:
feat: added pokemon api model
- added getPokemonList function

Miqqy0069 committed:
feat: added pokemon model
- pr feedback
- removed @Default

Miqqy0069 committed:
feat: added pokemon model
- removed pokemonDetails on main

shawn-ffufm commented on lib/api/handler/pokemon_api.dart:
```diff
@@ -9,5 +13,19 @@ class PokemonApi {
+    required String limit,
+  }) async {
+    final queryParameters = <String, dynamic>{};
+    queryParameters[queryParamLimit] = limit;
```
I don't see anything related to your `queryParamLimit` under your string_constant.dart file, is this even existing? 🤔

shawn-ffufm commented on lib/api/handler/pokemon_api.dart:
```diff
@@ -9,5 +13,19 @@ class PokemonApi {
+  }) async {
+    final queryParameters = <String, dynamic>{};
+    queryParameters[queryParamLimit] = limit;
+    queryParameters[queryParamOffset] = offset;
```
same here

shawn-ffufm commented on lib/api/handler/pokemon_api.dart:
```diff
@@ -9,5 +13,19 @@ class PokemonApi {
+    queryParameters[queryParamLimit] = limit;
+    queryParameters[queryParamOffset] = offset;
+    final uri = baseUri.replace(
+        queryParameters: queryParameters, path: '${baseUri.path}$endPointPokemon');
```
I've mentioned this already in your previous pr - again do you need the whitespace here? else remove it 🙂

shawn-ffufm reviewed (changes requested):
see pr feedback 🙂

Miqqy0069 committed:
feat: added pokemon api model
- pr feedback
- removed constants for query parameters
- removed unnecessary white spaces

Miqqy0069 committed:
feat: added pokemon api model
- pr feedback
- removed white space for line 22
